src/core/service.h

```diff
@@ -75,7 +75,8 @@ typedef enum ServiceResult {
         SERVICE_FAILURE_CORE_DUMP,
         SERVICE_FAILURE_WATCHDOG,
         SERVICE_FAILURE_START_LIMIT_HIT,
-        SERVICE_FAILURE_OOM_KILL,
+        SERVICE_FAILURE_OOM_KILL, /* OOM Kill by the Kernel */
+        SERVICE_FAILURE_OOMD_KILL, /* OOM Kill by systemd-oomd */
```

Comment:
hmm, oomd is a specific piece of software, not a condition or cause. I'd name this SERVICE_FAILURE_MANAGED_OOM_KILL, i.e. emphasize that this is a *managed* kill, regardless of the fact this is implemented in oomd.

I wonder though: why distinguish this from regular OOM kills? In which way is that relevant to UIs? isn't this just an implementation detail we should abstract away?

or in other words, is this distinction something we should expose in API or just in logs?

Reply:
A separate failure state might be useful for troubleshooting, as mentioned by @Werkov in https://github.com/systemd/systemd/issues/20649#issuecomment-965177073

Reply:
not sure I am convinced. I mean, making troubleshooting easy is a worthy goal, but to me the answer for that is good logging, not propagating the implementation detail here...